modules/restore-replace-stopped-etcd-member.adoc

```diff
@@ -0,0 +1,131 @@
+https://10.0.164.97:2379 is healthy: successfully committed proposal: took = 16.621645ms
+----
++
+You can now exit the node shell.
+
+. Delete and recreate the master machine. After this machine is recreated, a new revision is forced and etcd scales up automatically.
++
+If you are running an installer-provision installation, or you used the Machine API to create your machines, follow these steps. Otherwise, you must create the new master using the same method that was used to originally create it.
+
+.. Obtain the machine for the unhealthy member.
++
+In a terminal that has access to the cluster as a `cluster-admin` user, run the following command:
++
+----
+$ oc get machines -n openshift-machine-api -o wide
+
+NAME                                        PHASE     TYPE        REGION      ZONE         AGE     NODE                           PROVIDERID                              STATE
+clustername-8qw5l-master-0                  Running   m4.xlarge   us-east-1   us-east-1a   3h37m   ip-10-0-131-183.ec2.internal   aws:///us-east-1a/i-0ec2782f8287dfb7e   stopped <1>
+clustername-8qw5l-master-1                  Running   m4.xlarge   us-east-1   us-east-1b   3h37m   ip-10-0-154-204.ec2.internal   aws:///us-east-1b/i-096c349b700a19631   running
+clustername-8qw5l-master-2                  Running   m4.xlarge   us-east-1   us-east-1c   3h37m   ip-10-0-164-97.ec2.internal    aws:///us-east-1c/i-02626f1dba9ed5bba   running
+clustername-8qw5l-worker-us-east-1a-wbtgd   Running   m4.large    us-east-1   us-east-1a   3h28m   ip-10-0-129-226.ec2.internal   aws:///us-east-1a/i-010ef6279b4662ced   running
+clustername-8qw5l-worker-us-east-1b-lrdxb   Running   m4.large    us-east-1   us-east-1b   3h28m   ip-10-0-144-248.ec2.internal   aws:///us-east-1b/i-0cb45ac45a166173b   running
+clustername-8qw5l-worker-us-east-1c-pkg26   Running   m4.large    us-east-1   us-east-1c   3h28m   ip-10-0-170-181.ec2.internal   aws:///us-east-1c/i-06861c00007751b0a   running
+----
+<1> This is the master machine for the unhealthy node, `ip-10-0-131-183.ec2.internal`.
+
+.. Save the machine configuration to your local file system.
++
+----
+$ oc get machine clustername-8qw5l-master-0 -n openshift-machine-api -o yaml > new-master-machine.yaml
+----
+
+.. Edit the `new-master-machine.yaml` file.
+
+... Remove the entire `status` section.
+
+... Remove the `providerID` field.
+
+... Change the `name` field to a new name.
++
+It is recommended to keep the same base name as the old machine and change the ending number to the next available number. In this example, `clustername-8qw5l-master-0` is changed to `clustername-8qw5l-master-3`.
+
+... Update the `selfLink` field to use the new machine name from the previous step.
+
+.. Delete the machine of the unhealthy member:
++
+----
+$ oc delete machine ci-ln-mpprd6t-d5d6b-n94jk-master-0 -n openshift-machine-api
+----
++
+// TODO: This command takes a few minutes to come back, just wait it out?
```

Comment:
@alaypatel07 Deleting the machine takes a while. It's a few minutes before the prompt is returned to you in the terminal. Should we tell users to wait until the prompt is returned before proceeding to the next step? Should we tell them to wait until the machine is actually deleted (which is a few more minutes past the prompt returning) before going to the next step?

Reply:
@bergerhoffer yes, this is expected as the VM is being deleted under the hood. 

> Should we tell users to wait until the prompt is returned before proceeding to the next step?
yes, I think not only we should tell users to wait for the prompt to return, but I know some anxious ones like me will just ctrl+C out of it. So we should ask users to wait until  

`oc get machine ci-ln-mpprd6t-d5d6b-n94jk-master-0 -n openshift-machine-api` returns "No resources fouond"

make be even automate that with:

`$ until $(oc get machine ci-ln-mpprd6t-d5d6b-n94jk-master-0 -n openshift-machine-api | grep "No resources found"); do sleep 1; echo "waiting for the machine to be deleted"; done` 

as the next step after deletion.

Reply:
Thanks, I'll add and will test it out tomorrow.

Reply:
@alaypatel07 I tried this just now with a machine that doesn't exist, and I get `Error from server (NotFound): machines.machine.openshift.io "ci-ln-mpprd6t-d5d6b-n94jk-master-0" not found`

I haven't tested with an actual machine I've deleted, but if the message is the same, this until command doesn't work.

Reply:
I don't know if this will help, but you can use the `--wait` option for `oc`:

```
$ oc delete pod sctpserver --wait
pod "sctpserver" deleted
... prompt waits until success ...
$ 
```

Perhaps this works for the Machine resource as well?